inc/frontend/namespace.php

```diff
@@ -59,7 +59,25 @@ function output_script() {
 function mangle_images( $content ) {
 	// Find images
 	$searcher = '#<img[^>]+(?:class=[\'"]([^\'"]*wp-image-(\d+)[^\'"]*)|data-gaussholder-id="(\d+)")[^>]+>#x';
-	if ( ! preg_match_all( $searcher, $content, $images, PREG_SET_ORDER ) ) {
+	$preg_match_result = preg_match_all( $searcher, $content, $images, PREG_SET_ORDER );
+
+	/**
+	 * Filter the regexp results when looking for images in a post content.
+	 *
+	 * By default, Gaussholder applies the $searcher regexp inside the_content filter callback. Some page builders
+	 * manage images in different ways so the result could be false.
+	 *
+	 * This filter allows to change that result but also the images list generated by preg_match_all( $searcher, $content, $images, PREG_SET_ORDER )
+	 * The $images parameter must be returned with the same format. That is:
+	 *
+	 * [
+	 *  0 => Image tag (<img class="..." src="..." ... />)
+	 *  1 => Image tag class
+	 *  2 => Attachment ID
+	 * ]
+	 */
+	$preg_match_result = apply_filters_ref_array( 'gaussholder.mangle_images_regexp_results', array( $preg_match_result, &$images, $content, $searcher ) );
```

Comment:
I think the `&$` will break in PHP 7.4 and shouldn't really be needed any more. It's best to declare the variable first, then let the function modify it.

Reply:
@roborourke I just tested it it 7.4 and it would fail outside an array but I'm using here `apply_filters_ref_array` and passing an array so I'm not seeing any error. Otherwise, the images list could not be changed.

Reply:
Ah ok, I thought it would fatal or at least chuck out a warning. I guess we're a ways off adopting PHP 7.4 anyway so 👍  - thanks for confirming it